oracle/src/oracleimporter.cpp

```diff
@@ -318,6 +322,13 @@ int OracleImporter::importTextSpoiler(CardSetPtr set, const QVariant &data)
     return cards;
 }
 
+void OracleImporter::sortColors(QStringList &colors)
+{
+    const QMap<QString, unsigned int> colorOrder{{"W", 0}, {"U", 1}, {"B", 2}, {"R", 3}, {"G", 4}};
+    std::sort(colors.begin(), colors.end(),
+              [&colorOrder](const QString a, const QString b) { return colorOrder[a] < colorOrder[b]; });
```

Comment:
Does this work if we add a random new color?

Reply:
No, it will crash. But for that to happen, this new color would have to be added to the mtgjson file itself. 
And since the sorting only operates on the output of the `OracleImporter::extractColors` method, that would also have to be updated to process this new color before it can cause a crash. If you'd like I can add a few lines to check whether the color is among the supported color codes, but I can't really imagine a new color being added either short or long term.

Reply:
Can you add a fallback just in case? I wouldn't rule out a bug in mtgjson generating bad colors in the future.

Reply:
Added it.
I was wrong though, it wouldn't have crashed with new colors. Turns out QMap returns a default value (0 in this case) when it encounters an unknown key. This would have made the new color the same priority as "W".